PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:4 + add:0, change matches:
-    _inherit = ['pos.config']
-    _inherit = ['pos.order']
-    _inherit = ['pos.order.line']
-    _inherit = ['pos.session']

Total Changes: 14

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/pos_event/models/event_event.py
+++ b/addons/pos_event/models/event_event.py
@@ -2,6 +2,7 @@ from odoo import api, fields, models
 
 
 class EventEvent(models.Model):
+    _name = 'event.event'
     _inherit = ['event.event', 'pos.load.mixin']
 
     image_1024 = fields.Image("PoS Image", max_width=1024, max_height=1024)

--- a/addons/pos_event/models/event_question.py
+++ b/addons/pos_event/models/event_question.py
@@ -4,6 +4,7 @@ from odoo import api, models
 
 
 class EventQuestion(models.Model):
+    _name = 'event.question'
     _inherit = ['event.question', 'pos.load.mixin']
 
     @api.model

--- a/addons/pos_event/models/event_question_answer.py
+++ b/addons/pos_event/models/event_question_answer.py
@@ -4,6 +4,7 @@ from odoo import api, models
 
 
 class EventQuestionAnswer(models.Model):
+    _name = 'event.question.answer'
     _inherit = ['event.question.answer', 'pos.load.mixin']
 
     @api.model

--- a/addons/pos_event/models/event_registration.py
+++ b/addons/pos_event/models/event_registration.py
@@ -3,6 +3,7 @@ from odoo import fields, models, api
 
 
 class EventRegistration(models.Model):
+    _name = 'event.registration'
     _inherit = ['event.registration', 'pos.load.mixin']
 
     pos_order_id = fields.Many2one(related='pos_order_line_id.order_id', string='PoS Order')

--- a/addons/pos_event/models/event_registration_answer.py
+++ b/addons/pos_event/models/event_registration_answer.py
@@ -4,6 +4,7 @@ from odoo import api, models
 
 
 class EventRegistrationAnswer(models.Model):
+    _name = 'event.registration.answer'
     _inherit = ['event.registration.answer', 'pos.load.mixin']
 
     @api.model

--- a/addons/pos_event/models/event_ticket.py
+++ b/addons/pos_event/models/event_ticket.py
@@ -3,6 +3,7 @@ from odoo import models, api, fields
 
 
 class EventEventTicket(models.Model):
+    _name = 'event.event.ticket'
     _inherit = ['event.event.ticket', 'pos.load.mixin']
 
     @api.model

--- a/addons/pos_event/models/pos_config.py
+++ b/addons/pos_event/models/pos_config.py
@@ -3,7 +3,7 @@ from odoo import models
 
 
 class PosConfig(models.Model):
-    _inherit = ['pos.config']
+    _inherit = 'pos.config'
 
     def _update_events_seats(self, events):
         data = []

--- a/addons/pos_event/models/pos_order.py
+++ b/addons/pos_event/models/pos_order.py
@@ -3,7 +3,7 @@ from odoo import models, fields, api
 
 
 class PosOrder(models.Model):
-    _inherit = ['pos.order']
+    _inherit = 'pos.order'
 
     attendee_count = fields.Integer('Attendee Count', compute='_compute_attendee_count')
 

--- a/addons/pos_event/models/pos_order_line.py
+++ b/addons/pos_event/models/pos_order_line.py
@@ -3,7 +3,7 @@ from odoo import models, fields, api
 
 
 class PosOrderLine(models.Model):
-    _inherit = ['pos.order.line']
+    _inherit = 'pos.order.line'
 
     event_ticket_id = fields.Many2one('event.event.ticket', string='Event Ticket')
     event_registration_ids = fields.One2many('event.registration', 'pos_order_line_id', string='Event Registrations')

--- a/addons/pos_event/models/pos_session.py
+++ b/addons/pos_event/models/pos_session.py
@@ -3,7 +3,7 @@ from odoo import api, models
 
 
 class PosSession(models.Model):
-    _inherit = ['pos.session']
+    _inherit = 'pos.session'
 
     @api.model
     def _load_pos_data_models(self, config_id):
